src/api/app/controllers/webui/users/tokens_controller.rb

```diff
@@ -80,13 +78,17 @@ def set_token
   end
 
   def set_params
-    @params = params.except(:project_name, :package_name).require(:token).except(:string_readonly).permit(:type, :scm_token).tap do |token_parameters|
+    @params = params.except(:project_name, :package_name).require(:token).except(:string_readonly).permit(:type, :name, :scm_token).tap do |token_parameters|
       token_parameters.require(:type)
     end
     @params = @params.except(:scm_token) unless @params[:type] == 'workflow'
     @extra_params = params.slice(:project_name, :package_name).permit!
   end
 
+  def update_parameters
+    params.require(:token).except(:string_readonly).permit(:name, :scm_token).reject! { |k, v| k == 'scm_token' && v.empty? }
```

Comment:
It should be possible to update `scm_token` only if the token's type is `workflow`.

Reply:
Yes. I added that check in the `reject!` block.